src/pages/api/movies/[id].ts

```diff
@@ -0,0 +1,12 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import mongoose from "mongoose"
+import {movies} from "../../../server/models"
+
+mongoose.connect(process.env.DB_URL!) 
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+    const id = req.query.id
+    const data = await movies.find();
+    const movie = data.find(movie => movie.id === id)
```

Comment:
These two lines download all the movies from the database (`movies.find()`) and then loops through them to find the one that matches the requested ID (`data.find()`, which is  `Array.prototype.find()`).

If you're only going to use one, and you know how to find it, you could download just that single movie from the database, which will be faster – especially as the number of movies in the database grows.

The `find()` method on Mongoose models can take a "query" object that you could use for that. See the documentation for details.